codecov.yml

```diff
@@ -1,2 +1,23 @@
+# Documentation: https://docs.codecov.io/docs/codecov-yaml
 ignore:
   - "api/v1alpha1/zz_generated.deepcopy.go" # generated file
+
+# general codecov settings
+codecov:
+    notify:
+        require_ci_to_pass: false
+coverage:
+    status:
+        # sets the expected status for `codecov/project` check.
+        # `target: auto` means it will use the current coverage to calculate against
+        #          the give PR/commit.
+        # theshould is the max delta allowed when compared with the current coverage
+        project:
+            default:
+                target:  auto
+                threshold: 3%
```

Comment:
so what does this mean? we fail the check unless we increase the code coverage by 3%?

Reply:
This is indeed confusing.
The target is the number that defines the coverage we want. _Auto_ means codecov will calculate it by itself (the default). The threshold is the amount in percent that it can miss from the target. If the target is 80%, it will pass starting from 77%. It happens that at each push (because of codecov characteristics) the coverage result is flaky. This is basically to avoid this.